### ISSUE TITLE
Sometimes terraform marks virtual guests as created but actually there are still pending transactions

### ISSUE THREAD

**Issue by bondyk** [softlayer/terraform-provider-softlayer#47]
Sometimes it happens that terraform completes job too early saying that virtual guests are created but actually there are still pending transactions (can be checked via SoftLayer web UI) and of course the hosts are unreachable via ssh.
Usually it takes about 20 minutes to create Virtual Guests but sometimes terraform completes the job in a minute or so.

Normal process:
```
softlayer_virtual_guest.data-node: Still creating... (10s elapsed)
softlayer_virtual_guest.console: Still creating... (10s elapsed)
...
softlayer_virtual_guest.console: Still creating... (19m30s elapsed)
softlayer_virtual_guest.data-node: Still creating... (19m30s elapsed)
softlayer_virtual_guest.console: Still creating... (19m40s elapsed)
softlayer_virtual_guest.data-node: Still creating... (19m40s elapsed)
softlayer_virtual_guest.console: Creation complete
softlayer_virtual_guest.data-node: Still creating... (19m50s elapsed)
softlayer_virtual_guest.data-node: Still creating... (20m0s elapsed)
softlayer_virtual_guest.data-node: Still creating... (20m10s elapsed)
softlayer_virtual_guest.data-node: Creation complete
```
Abnormal process:
```
softlayer_virtual_guest.data-node: Still creating... (10s elapsed)
softlayer_virtual_guest.console: Still creating... (10s elapsed)
softlayer_virtual_guest.data-node: Still creating... (20s elapsed)
softlayer_virtual_guest.console: Still creating... (20s elapsed)
softlayer_virtual_guest.data-node: Still creating... (30s elapsed)
softlayer_virtual_guest.console: Still creating... (30s elapsed)
softlayer_virtual_guest.data-node: Still creating... (40s elapsed)
softlayer_virtual_guest.console: Still creating... (40s elapsed)
softlayer_virtual_guest.data-node: Still creating... (50s elapsed)
softlayer_virtual_guest.console: Still creating... (50s elapsed)
softlayer_virtual_guest.data-node: Still creating... (1m0s elapsed)
softlayer_virtual_guest.console: Still creating... (1m0s elapsed)
softlayer_virtual_guest.data-node: Still creating... (1m10s elapsed)
softlayer_virtual_guest.console: Still creating... (1m10s elapsed)
softlayer_virtual_guest.console: Creation complete
softlayer_virtual_guest.data-node: Creation complete

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

The state of your infrastructure has been saved to the path
below. This state is required to modify and destroy your
infrastructure, so keep it safe. To inspect the complete state
use the `terraform show` command.
```

In both cases was used the same terraform config but with changed host names only.
Uses terraform-provider-softlayer of version 1.2.0

**Comment by renier** [comment]
@bondyk Are you using the released terraform binary version 0.7.x? Are you using REST or XMLRPC endpoint? (check your ~/.softlayer; default is REST)

**Comment by bondyk** [comment]
@renier terraform of version 0.7.5. and default endpoint. BTW I don't see file ~/.softlayer

**Comment by bondyk** [comment]
Here is terraform config file: [softlayer.txt](https://github.com/softlayer/terraform-provider-softlayer/files/583966/softlayer.txt)

**Comment by renier** [comment]
@bondyk Are you comfortable building the provider binary from source? basically `make bin`.

**Comment by bondyk** [comment]
@renier yes, no problem. I was building from git repo until I found that there is a possibility to download binaries directly :)

**Comment by renier** [comment]
@bondyk So just playing a hunch. I'm thinking that the API is misbehaving and saying there are no active transactions to the provider when they still are. So what I'm doing is making sure we have two consecutive reports of the API saying there are no active transaction (each check is spaced by 10 seconds). You'll have to checkout the _no-active-transx_ branch to try it.

**Comment by bondyk** [comment]
@renier What should I see when run it? Additional log messages?

**Comment by renier** [comment]
@bondyk No. You would just see if you can still reproduce the situation

**Comment by bondyk** [comment]
@renier ok, I'll be testing it for next couple of days and will let you know about results. Thanks.

**Comment by renier** [comment]
We think this was due to the API misbehaving yesterday. We haven't seen this occur again. Closing, but feel free to reopen if you can reproduce.

**Comment by bondyk** [comment]
@renier The issue isn't reproducing anymore so I think you can close it

**Comment by bondyk** [comment]
@renier ooops, it's closed already, sorry. But anyway I confirm that it is fixed

**Comment by bondyk** [comment]
@renier When are you going to release a new version containing this fix?

**Comment by renier** [comment]
@bondyk Look for 1.3.0 very soon

**Comment by bondyk** [comment]
@renier It seems the fix is not included in release 1.3.0. Do you want to test it more before merging?

**Comment by renier** [comment]
@bondyk Right. The branch still exists, but it is not included in master yet. I think adding this additional delay in our code is not necessary anymore since it has been fixed on the SoftLayer API side.
